test/form/Form.spec.jsx

```diff
@@ -1,53 +1,117 @@
-import Bluebird from 'bluebird'
-import Immutable from 'immutable'
-import React from 'react'
-import TestUtils from 'react-testutils-additions'
-import sinon from 'sinon'
-import expect from 'unexpected'
-
-import ErrorMessage from '../../src/form/ErrorMessage'
-import Form from '../../src/form/Form'
-import {PropsSetter} from '../reactHelpers'
-import TestField from './TestField'
-
-function render (props) {
+import Bluebird from "bluebird";
+import Immutable from "immutable";
+import PropTypes from "prop-types";
+import React from "react";
+import TestUtils from "react-testutils-additions";
+import sinon from "sinon";
+import expect from "unexpected";
+
+import ErrorMessage from "../../src/form/ErrorMessage";
+import Form from "../../src/form/Form";
+import { PropsSetter } from "../reactHelpers";
+import TestField from "./TestField";
+
+// to be able to output in dom server errors this component was made
+class ServerErrorMessage extends React.Component {
+  static contextTypes = {
+    formValueScope: PropTypes.object.isRequired
+  };
+
+  static defaultProps = {
+    Component: "div"
+  };
+
+  static propTypes = {
+    name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    Component: PropTypes.any,
+    serverValidationErrors: PropTypes.object
+  };
+
+  parseName = name => {
+    // only split string names by dots, but keep non string names (for example number names
+    // like in IteratorField) as they are
+    return typeof name === "string" ? name.split(/\./g) : [name];
+  };
+
+  getError = name => {
+    const {
+      updatedServerValidationErrors,
+      props
+    } = this.context.formValueScope;
+
+    const { serverValidationErrors } = props;
+
+    if (serverValidationErrors) {
+      const specificError = updatedServerValidationErrors
+        ? updatedServerValidationErrors.getIn(this.parseName(name))
+        : serverValidationErrors.getIn(this.parseName(name));
+
+      return specificError;
+    }
+    return null;
+  };
+
+  render() {
+    const { name, Component, ...other } = this.props;
+    const extendedErrorMessage = this.getError(name);
+
+    return extendedErrorMessage ? (
+      <Component {...other}>{extendedErrorMessage}</Component>
+    ) : null;
+  }
+}
+
+function render(props) {
   const initialValue = Immutable.fromJS({
-    firstName: '',
-    lastName: '',
+    firstName: "",
+    lastName: "",
     address: {
-      street: ''
+      street: ""
```

Comment:
Sorry but I'll wait for a more readable diff before continuing...

Reply:
No probs, my fault